tensorboard/data/server/pip_package/build.py

```diff
@@ -75,16 +75,19 @@ def main():
     if args.universal:
         platform_tag = "any"
     else:
+        cpu_name = platform.machine()
+        mac_platform = "macosx_10_9"
+        if cpu_name == "arm64":
```

Comment:
Instead of checking this cpu automatically, could you add a flag to switch the cpu_name and platform? We did not have intent for the binary to work on arm64, it just happens to. With the flag passing in, it's pretty much at your own control on the command running level to do so.

Reply:
My only hesitation with doing this is that the binary you build likely only works on the platform/cpu you build it on. For instance, if someone is running on Linux x86_64, changing the wheel name to "arm64_maxosx_11_0" likely isn't going to produce a wheel that will work on M1, and vice versa. So it may seem misleading if the user can choose the platform/cpu themselves.

But if you still want me to do this, I'm happy to do it. Or if I'm misunderstanding something and these settings somehow control what gets built, let me know.

Reply:
Hmm, but if that were the case, then why would we be passing `-p <platform_tag` to `python setup.py bdist_wheel`? Maybe it does allow for cross compiling? Or maybe that only controls how old the macOS release version that gets supported is.

Before this PR, platform was dynamically chosen by which platform you're running on. Are you suggesting to make that a flag too?

Reply:
Out of curiosity, why does the package have cpu/platform-specific wheels AND universal wheels on PyPI? Does the package actually depend on the OS or is it pure-Python? If I'm only trying to build for myself, can I just build a universal wheel to avoid the issues I reported in #5713?

Reply:
Looks like the only differences between the wheels is whether or not they come with a `tensorboard_data_server/bin/server` binary.

Reply:
@japie1235813 any comments on the above?

Reply:
We might want to cross-compile building the binary (e.g. building arm64 binary on x86 env). We don't want to have build.py make the assumption of cpu/os of where the binary comes from. build.py also does not check it and has no control of how the binary is built. Adding a flag provides the needs of not assuming anything and makes it clear that it's on the user's will to build an arm64 wheel.

Reply:
Okay, so if I add those flags and run `python build.py --platform=macosx_11_0 --cpu=arm64` on a Linux x86_64 machine, it will produce a wheel that works on macOS arm64, not a wheel that works on Linux x86_64? I'm still confused as to whether or not `build.py` actually supports cross-compiling. If it does, then great, I'll add those flags. If it doesn't, then adding flags is useless since you can only build a wheel for your current system.

Reply:
@japie1235813 did you ever hear back from the rest of the team? If `build.py` doesn't support cross-compiling, then there's no reason to add a flag that lets users shoot themselves in the foot. If you specify a different `--platform` and `--cpu` than the one you are running on, `build.py` will happily build a wheel for that platform/cpu, but it won't run on that platform/cpu.

Reply:
The `build.py` checks if the env is supported (in which case officially we only support x86) but does not check anything about `server_binary` and have no idea if the `server_binary` is compatible with the supported cup/os.
In current scenario (hardcode x86) on your machine, it builds a wheel claiming supporting the x86 but actually only supports arm64. 

What you are suggesting will work if you're building an arm64 binary on arm64 machine, but still does not support cross-compilation. Technically you can bazel build arm64 Rustboard binary on x86 machine. Then in the build.py, adding the flag to specify the cpu that the binary is built for creates a wheel could be run on arm64.